src/main/java/com/jcabi/http/wire/LastModifiedCachingWire.java

```diff
@@ -0,0 +1,207 @@
+        final int read) throws IOException {
+        final Response rsp;
+        if (method.equals(Request.GET)) {
+            rsp = this.lookInCache(
+                req, home, method, headers, content, connect, read
+            );
+        } else {
+            rsp = this.origin.send(
+                req, home, method, headers, content, connect, read
+            );
+        }
+        return rsp;
+    }
+
+    /**
+     * Check cache and update if needed.
+     *
+     * @param req Request
+     * @param home URI to fetch
+     * @param method HTTP method
+     * @param headers Headers
+     * @param content HTTP body
+     * @param connect The connect timeout
+     * @param read The read timeout
+     * @return Response obtained
+     * @throws IOException if fails
+     * @checkstyle ParameterNumber (6 lines)
+     */
+    private Response lookInCache(final Request req,
+        final String home,
+        final String method,
+        final Collection<Map.Entry<String, String>> headers,
+        final InputStream content,
+        final int connect,
+        final int read) throws IOException {
+        final Response rspReceived;
+        if (this.cache.containsKey(req)) {
+            rspReceived = this.updateCache(
+                req, home, method, headers, content, connect, read
+            );
+        } else {
+            rspReceived = this.origin.send(
+                req, home, method, headers, content, connect, read
+            );
+            this.addToCache(req, rspReceived);
+        }
+        return rspReceived;
+    }
+
+    /**
+     * Check response and update cache if needed.
+     *
+     * @param req Request
+     * @param home URI to fetch
+     * @param method HTTP method
+     * @param headers Headers
+     * @param content HTTP body
+     * @param connect The connect timeout
+     * @param read The read timeout
+     * @return Response obtained
+     * @throws IOException if fails
+     * @checkstyle ParameterNumber (8 lines)
+     */
+    private Response updateCache(final Request req,
+        final String home,
+        final String method,
+        final Collection<Map.Entry<String, String>> headers,
+        final InputStream content,
+        final int connect,
+        final int read) throws IOException {
+        final Response rspCashed = this.cache.get(req);
+        final Collection<Map.Entry<String, String>> hdrs = this.enrich(
+            headers, rspCashed
+        );
+        Response rsp = this.origin.send(
+            req, home, method, hdrs, content, connect, read
+        );
+        if (rsp.status() == HttpURLConnection.HTTP_NOT_MODIFIED) {
+            rsp = rspCashed;
+        } else {
+            this.addToCache(req, rsp);
+        }
+        return rsp;
+    }
+
+    /**
+     * Add response to cache.
+     *  @param req The key of the request
+     * @param rsp The response to add
+     */
+    private void addToCache(final Request req, final Response rsp) {
```

Comment:
Given that this may update an existing entry in the cache, using `add` in the name seems somewhat confusing.